inductiva/coastal/coastal_area.py

```diff
@@ -126,6 +128,39 @@ def get_config_filename(self, simulator: simulators.Simulator):
     def create_input_files(self, simulator: simulators.Simulator):
         pass
 
+    def _check_valid_wave(self):
+        """Checks that the wave amplitude is valid.
+
+        Raises:
+            ValueError: If the wave amplitude is not valid.
+        """
+
+        depths_grid = self.bathymetry.depths_grid()
+
+        if self.wave_source_location == "W":
+            wave_boundary = depths_grid[0, :]
+
+        elif self.wave_source_location == "E":
+            wave_boundary = depths_grid[-1, :]
+
+        elif self.wave_source_location == "N":
+            wave_boundary = depths_grid[:, -1]
+
+        elif self.wave_source_location == "S":
+            wave_boundary = depths_grid[:, 0]
+
+        else:
+            raise ValueError("Invalid wave source location.")
+
+        #The greater the value of depth, the deeper the water, so
+        #we want to check the minimum depth using the min value
+
+        minimum_depth = np.min(wave_boundary)
+        if self.wave_amplitude > minimum_depth:
```

Comment:
I think this kind of catches the error I mentioned above - Let's think that we have a bitch on the wave source location, then the minimum_depth will be negative. Which implies all wave_amplitudes will raise an error.

However, the error below is confusing because of the depths of the bathymetry (positive and negative signs), do you want to rephrase slightly so it is more intuitive what is going on.

Reply:
Great, but I don't think that is super explicit for some one using this for the first times. I actually like a lot more your first line: raises an error if the amplitude exceeds the minimum depth or if the wave source is in the beach - clean and simple.

Reply:
Chatting with @sarmento we found out that there is a bunch of pathological cases that we dont yet know if fall in one of the buckets. His suggestion is to mantain the error as is (an objective truth) and suggest to the user the possible causes for it. This leads the user to inspect the bathymetry and make the right decision. In the future we can implement this if we understand the pathological cases.

Reply:
So it will be something like: "Wave amplitude is larger than the "
                              "minimum depth of the bathymetry in the "
                              "boundary where waves are generated.
This can be either due to a very high wave amplitude or
ill defined wave source direction."